### ISSUE TITLE
unblockPlacements concurrent calls will raise panic bug

### ISSUE THREAD

**Issue by CodingSinger** [dapr/dapr#3364]
## In what area(s)?

/area placement

## What version of Dapr?

master branch 

## Expected Behavior

```
func (p *ActorPlacement) onPlacementOrder(in *v1pb.PlacementOrder) {
	log.Debugf("placement order received: %s", in.Operation)
	diag.DefaultMonitoring.ActorPlacementTableOperationReceived(in.Operation)

	// lock all incoming calls when an updated table arrives
	p.operationUpdateLock.Lock()
	defer p.operationUpdateLock.Unlock()

	switch in.Operation {
	case lockOperation:
		p.blockPlacements()

		go func() {
			// TODO: Use lock-free table update.
			// current implementation is distributed two-phase locking algorithm.
			// If placement experiences intermittently outage during updateplacement,
			// user application will face 5 second blocking even if it can avoid deadlock.
			// It can impact the entire system.
			time.Sleep(time.Second * 5)
			p.unblockPlacements()
		}()

	case unlockOperation:
		p.unblockPlacements()

	case updateOperation:
		p.updatePlacements(in.Tables)
	}
}
```

`unblockPlacements` Will be called concurrently. 

When the `unblockPlacements` method is called concurrently, the program is normal.

## Actual Behavior

program panic

```
=== RUN   TestConcurrentUnblockPlacements
=== RUN   TestConcurrentUnblockPlacements/concurrent_unlock
panic: close of closed channel

goroutine 144 [running]:
github.com/dapr/dapr/pkg/actors/internal.(*ActorPlacement).unblockPlacements(...)
	/Users/go/src/github.com/dapr/dapr/pkg/actors/internal/placement.go:347
github.com/dapr/dapr/pkg/actors/internal.TestConcurrentUnblockPlacements.func3.1(0xc000454d00)
	/Users/go/src/github.com/dapr/dapr/pkg/actors/internal/placement_test.go:264 +0x4c
created by github.com/dapr/dapr/pkg/actors/internal.TestConcurrentUnblockPlacements.func3
	/Users/go/src/github.com/dapr/dapr/pkg/actors/internal/placement_test.go:263 +0x50

Process finished with exit code 1

```

## Steps to Reproduce the Problem

```
func TestConcurrentUnblockPlacements(t *testing.T) {
	appHealthFunc := func() bool { return true }
	tableUpdateFunc := func() {}
	testPlacement := NewActorPlacement(
		[]string{}, nil,
		"testAppID", "127.0.0.1:1000",
		[]string{"actorOne", "actorTwo"},
		appHealthFunc, tableUpdateFunc)

	t.Run("concurrent_unlock", func(t *testing.T) {
		for i := 0; i < 10000; i++ {
			testPlacement.blockPlacements()
			wg := sync.WaitGroup{}
			wg.Add(2)
			go func() {
				testPlacement.unblockPlacements()
				wg.Done()
			}()
			go func() {
				testPlacement.unblockPlacements()
				wg.Done()
			}()
			wg.Wait()
		}
	})
}
```

**Comment by CodingSinger** [comment]
I submitted an MR to fix this problem. #3365

**Comment by yaron2** [comment]
Thanks for this @CodingSinger

**Comment by yaron2** [comment]
Closed with #3365
